pkg/cableengine/ipsec/ipsec.go

```diff
@@ -95,7 +95,30 @@ func (i *Engine) StartEngine(ignition bool) error {
 		forwardToSubPostroutingRuleSpec := []string{"-j", "SUBMARINER-POSTROUTING"}
 		ipt.AppendUnique("nat", "POSTROUTING", forwardToSubPostroutingRuleSpec...)
 		forwardToSubForwardRuleSpec := []string{"-j", "SUBMARINER-FORWARD"}
-		ipt.AppendUnique("filter", "FORWARD", forwardToSubForwardRuleSpec...)
+		rules, err := ipt.List("filter", "FORWARD")
+		if err != nil {
+			klog.Fatalf("error listing the rules in FORWARD chain: %v", err)
+		}
+
+		appendAt := len(rules) + 1
+		insertAt := appendAt
+		for i, rule := range rules {
+			if rule == "-A FORWARD -j SUBMARINER-FORWARD" {
+				insertAt = -1
+				break
+			} else if rule == "-A FORWARD -j REJECT --reject-with icmp-host-prohibited" {
+				insertAt = i
+				break
+			}
+		}
+
+		if insertAt == appendAt {
+			// Append the rule at the end of the FORWARD Chain.
+			ipt.AppendUnique("filter", "FORWARD", forwardToSubForwardRuleSpec...)
```

Comment:
Since we've already checked if the entry already exists, it seems we could just use Append here to avoid the duplicate checking in AppendUnique but I assume it's relatively negligible.